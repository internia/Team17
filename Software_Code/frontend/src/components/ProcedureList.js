import React, { Component } from 'react';
import Card from './Card'
// import { watchFile } from 'fs';
// import PropTypes from 'prop-types'

class ProcedureList extends Component {
  render() {
    return this.props.procedures.map((card) => (
        <Card key={card.id} card={card} />
    ));
  }

}

// ProcedureList.propTypes = {
//   cards: PropTypes.array.isRequired
// }

export default ProcedureList;
