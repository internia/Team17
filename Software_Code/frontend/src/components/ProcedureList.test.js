import React from 'react';
import { render } from '@testing-library/react';
import ProcedureList from './ProcedureList';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { isCompositeComponent } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() })


it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<ProcedureList procedures={[{"ProviderID":450864,"ProviderName":"BAYLOR SCOTT & WHITE TEXAS SPINE & JOINT HOSPITAL","Discharges":11,"CoveredCharges":61518.1,"TotalPayments":6546,"MedicarePayments":5235.09,"GPDID":520,
    "DRGDefinition":"BACK & NECK PROC EXC SPINAL FUSION W/O CC/MCC","Latitude":32.3295,"longitude":-95.2985,"Address":"1814 ROSELAND BOULEVARD","City":"TYLER","RefferalRegion":"TX - Tyler","State":"TX","Zip":75701},
    {"ProviderID":450856,"ProviderName":"SOUTH TEXAS SPINE AND SURGICAL HOSPITAL","Discharges":17,"CoveredCharges":24517.9,"TotalPayments":6702.82,"MedicarePayments":5133.06,"GPDID":520,
    "DRGDefinition":"BACK & NECK PROC EXC SPINAL FUSION W/O CC/MCC","Latitude":29.6139,"longitude":-98.4776,"Address":"18600 NORTH HARDY OAK BLVD","City":"SAN ANTONIO","RefferalRegion":"TX - San Antonio","State":"TX","Zip":78258}]}></ProcedureList>, div)
})

it('renders searchbar correctly', () => {
    function callBack() {
        console.log("called back")
    }


    render(<ProcedureList procedures={[{"ProviderID":450864,"ProviderName":"BAYLOR SCOTT & WHITE TEXAS SPINE & JOINT HOSPITAL","Discharges":11,"CoveredCharges":61518.1,"TotalPayments":6546,"MedicarePayments":5235.09,"GPDID":520,
    "DRGDefinition":"BACK & NECK PROC EXC SPINAL FUSION W/O CC/MCC","Latitude":32.3295,"longitude":-95.2985,"Address":"1814 ROSELAND BOULEVARD","City":"TYLER","RefferalRegion":"TX - Tyler","State":"TX","Zip":75701},
    {"ProviderID":450856,"ProviderName":"SOUTH TEXAS SPINE AND SURGICAL HOSPITAL","Discharges":17,"CoveredCharges":24517.9,"TotalPayments":6702.82,"MedicarePayments":5133.06,"GPDID":520,
    "DRGDefinition":"BACK & NECK PROC EXC SPINAL FUSION W/O CC/MCC","Latitude":29.6139,"longitude":-98.4776,"Address":"18600 NORTH HARDY OAK BLVD","City":"SAN ANTONIO","RefferalRegion":"TX - San Antonio","State":"TX","Zip":78258}]}></ProcedureList>)
})


it("matches snapshot", () => {
    function callBack() {
        console.log("called back")
    }
    const tree = renderer.create(<ProcedureList procedures={[{"ProviderID":450864,"ProviderName":"BAYLOR SCOTT & WHITE TEXAS SPINE & JOINT HOSPITAL","Discharges":11,"CoveredCharges":61518.1,"TotalPayments":6546,"MedicarePayments":5235.09,"GPDID":520,
    "DRGDefinition":"BACK & NECK PROC EXC SPINAL FUSION W/O CC/MCC","Latitude":32.3295,"longitude":-95.2985,"Address":"1814 ROSELAND BOULEVARD","City":"TYLER","RefferalRegion":"TX - Tyler","State":"TX","Zip":75701},
    {"ProviderID":450856,"ProviderName":"SOUTH TEXAS SPINE AND SURGICAL HOSPITAL","Discharges":17,"CoveredCharges":24517.9,"TotalPayments":6702.82,"MedicarePayments":5133.06,"GPDID":520,
    "DRGDefinition":"BACK & NECK PROC EXC SPINAL FUSION W/O CC/MCC","Latitude":29.6139,"longitude":-98.4776,"Address":"18600 NORTH HARDY OAK BLVD","City":"SAN ANTONIO","RefferalRegion":"TX - San Antonio","State":"TX","Zip":78258}]}></ProcedureList>)
    expect(tree).toMatchSnapshot();
    // expect(true).toBeTruthy();
})

