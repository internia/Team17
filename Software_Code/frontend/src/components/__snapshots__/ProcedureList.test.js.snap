// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="sc-AykKC JXmus"
  >
    <div
      style={
        Object {
          "backgroundColor": "1#dcdde1",
          "border": "1px solid",
          "margin-bottom": "1em",
          "padding": "1em",
        }
      }
    >
      <div
        class="row row-cols-2"
        style={
          Object {
            "fontSize": "1.4em",
            "fontweight": 600,
          }
        }
      >
        <div
          class="col"
        >
           
          <b>
            BAYLOR SCOTT & WHITE TEXAS SPINE & JOINT HOSPITAL
          </b>
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          BACK & NECK PROC EXC SPINAL FUSION W/O CC/MCC
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          <b>
            GPDID Code: 
          </b>
          520
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          <b>
            Average Price: 
          </b>
        </div>
        <div
          class="col"
        >
          $
          6546
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          <b>
            Distance: 
          </b>
        </div>
        <div
          class="col"
        >
          256 Miles
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          <b>
            More information: 
          </b>
        </div>
        <div
          class="col"
        >
          BUTTON
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-AykKC JXmus"
  >
    <div
      style={
        Object {
          "backgroundColor": "1#dcdde1",
          "border": "1px solid",
          "margin-bottom": "1em",
          "padding": "1em",
        }
      }
    >
      <div
        class="row row-cols-2"
        style={
          Object {
            "fontSize": "1.4em",
            "fontweight": 600,
          }
        }
      >
        <div
          class="col"
        >
           
          <b>
            SOUTH TEXAS SPINE AND SURGICAL HOSPITAL
          </b>
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          BACK & NECK PROC EXC SPINAL FUSION W/O CC/MCC
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          <b>
            GPDID Code: 
          </b>
          520
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          <b>
            Average Price: 
          </b>
        </div>
        <div
          class="col"
        >
          $
          6702.82
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          <b>
            Distance: 
          </b>
        </div>
        <div
          class="col"
        >
          256 Miles
        </div>
      </div>
      <div
        class="row row-cols-2"
      >
        <div
          class="col"
        >
          <b>
            More information: 
          </b>
        </div>
        <div
          class="col"
        >
          BUTTON
        </div>
      </div>
    </div>
  </div>,
]
`;
